Alloy/MTLDevice+Extensions.swift

```diff
@@ -0,0 +1,53 @@
+//
+//  MTLDevice+Extensions.swift
+//  Alloy-iOS
+//
+//  Created by Vladimir Pavlov on 06/05/2019.
+//
+
+import Metal
+import CoreGraphics
+
+@available(iOS 11.0, macOS 10.12, *)
+public extension MTLDevice {
+
+    func maxTextureSize(desiredSize: CGSize) -> CGSize {
+        let maxSide: CGFloat
+        if self.supportsOnly8K() {
+            maxSide = 8192 * metalSafeMultiplier
+        } else {
+            maxSide = 16_384 * metalSafeMultiplier
+        }
+
+        guard desiredSize.width > 0, desiredSize.height > 0 else { return .zero }
+        let aspectRatio = desiredSize.width / desiredSize.height
+        if aspectRatio > 1 {
+            let resultWidth = min(desiredSize.width, maxSide)
+            let resultHeight = resultWidth / aspectRatio
+            return CGSize(width: resultWidth, height: resultHeight)
+        } else {
+            let resultHeight = min(desiredSize.height, maxSide)
+            let resultWidth = resultHeight * aspectRatio
+            return CGSize(width: resultWidth, height: resultHeight)
+        }
+    }
+
+    private func supportsOnly8K() -> Bool {
+        #if os(macOS)
+        return false
```

Comment:
that's temporary of an actual fact?

Reply:
All Mac versions that Alloy supports can create 16K textures.